PetrolEngine.cpp

```diff
@@ -2,19 +2,16 @@
 #include <iostream>
 
 PetrolEngine::PetrolEngine(int power, float capacity, int gears)
-    : power_(power)
-    , capacity_(capacity)
-    , gears_(gears)
-    , currentGear_(0)
-{
-    std::cout << __FUNCTION__ << std::endl;
+    : Engine(power), capacity_(capacity), gears_(gears), currentGear_(0) {
 }
 
-void PetrolEngine::changeGear(int gear)
-{
-    // TODO: Add checking if gear is between -1 and gears_
-    // -1 is for REAR
-    // 0 is for NEUTRAL
+void PetrolEngine::changeGear(int gear) {
+    if (-1 > gear || gears_ < gear) {
+        throw InvalidGear("Non existant gear");
+    }
     currentGear_ = gear;
```

Comment:
My suggestion. I suggest adding a case when the user wants to shift into reverse gear from a gear higher than neutral. It could damage the real gearbox haha! I suggest also to test it :}

Reply:
or other direction from -1 to 5